Source/WebCore/rendering/RenderLayerScrollableArea.cpp

```diff
@@ -847,14 +847,9 @@ bool RenderLayerScrollableArea::canShowNonOverlayScrollbars() const
 
 void RenderLayerScrollableArea::createScrollbarsController()
 {
-    if (usesAsyncScrolling()) {
-        if (auto scrollbarController = m_layer.page().chrome().client().createScrollbarsController(m_layer.page(), *this)) {
-            setScrollbarsController(WTFMove(scrollbarController));
-            return;
-        }
-    }
-
-    ScrollableArea::createScrollbarsController();
+    m_layer.page().chrome().client().ensureScrollbarsController(m_layer.page(), *this);
```

Comment:
If the client has a function called `ensureScrollbarsController()` then it should always make one, which means the default client implementation has to fall back to calling `ScrollableArea::createScrollbarsController()`. Or you call it something like `adjustScrollbarsController()` and allow it to leave the scrollbars controller or null it out.

Reply:
Right fixed.